process/factory/interceptorscontainer/metaInterceptorsContainerFactory.go

```diff
@@ -0,0 +1,337 @@
+package interceptorscontainer
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/core/throttler"
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/dataValidators"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/process/factory/containers"
+	processInterceptors "github.com/ElrondNetwork/elrond-go/process/interceptors"
+	interceptorFactory "github.com/ElrondNetwork/elrond-go/process/interceptors/factory"
+	"github.com/ElrondNetwork/elrond-go/process/interceptors/processor"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+var _ process.InterceptorsContainerFactory = (*metaInterceptorsContainerFactory)(nil)
+
+// metaInterceptorsContainerFactory will handle the creation the interceptors container for metachain
+type metaInterceptorsContainerFactory struct {
+	*baseInterceptorsContainerFactory
+}
+
+// NewMetaInterceptorsContainerFactory is responsible for creating a new interceptors factory object
+func NewMetaInterceptorsContainerFactory(
+	shardCoordinator sharding.Coordinator,
```

Comment:
an arg would reduce the dupplicate parameters

Reply:
done